src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -221,24 +263,176 @@ public void onTraversalStart() {
      */
     @Override
     public void traverse() {
-        try {
-            importer.importBatch();
-        } catch (IOException e) {
-            throw new UserException("GenomicsDB importBatch raised exception", e);
+
+        int updatedBatchSize = (batchSize == DEFAULT_ZERO_BATCH_SIZE) ? sampleNames.size() : batchSize;
+        int sampleCount = sampleNames.size();
+        int totalBatchCount = (sampleCount/updatedBatchSize) + (sampleCount%updatedBatchSize==0 ? 0 : 1);
+
+        GenomicsDBImporter importer;
+
+        for (int i = 0, batchCount = 1; i < sampleCount; i += updatedBatchSize, ++batchCount) {
+
+            Map<String, FeatureReader<VariantContext>> sampleToReaderMap =
+                getFeatureReaders(sampleNames, sampleNameToVcfUri, updatedBatchSize, sampleCount, i);
+
+            logger.info("Importing batch " + batchCount + " with " + sampleToReaderMap.size() + " samples");
+            final long variantContextBufferSize = vcfBufferSizePerSample * sampleToReaderMap.size();
+
+            GenomicsDBImportConfiguration.ImportConfiguration importConfiguration =
+                    createImportConfiguration(workspace, arrayName,
+                            variantContextBufferSize, segmentSize,
+                            "", "",
+                            i, (i+updatedBatchSize-1));
+
+            try {
+                importer = new GenomicsDBImporter(sampleToReaderMap, mergedHeader, intervals.get(0), importConfiguration);
+            } catch (IOException e) {
+                throw new UserException("Error initializing GenomicsDBImporter in batch " + batchCount, e);
+            }
+
+            try {
+                importer.importBatch();
+                assert importer.isDone();
```

Comment:
removed this assertion. this is redundant cause all exceptions are thrown inside importBatch()